sa/sa.go

```diff
@@ -2143,3 +2144,44 @@ func (ssa *SQLStorageAuthority) IncidentsForSerial(ctx context.Context, req *sap
 	}
 	return incidentsForSerial, nil
 }
+
+// SerialsForIncident queries the provided incident table and returns the
+// resulting rows as a stream of `*sapb.IncidentSerial`s. The caller receives
+// this stream of serials by iterating over the `Recv()` method of the returned
+// streaming client. An `io.EOF` error signals that there are no more serials to
+// send. If the incident table in question contains zero rows, only an `io.EOF`
+// error is returned.
+func (ssa *SQLStorageAuthority) SerialsForIncident(req *sapb.SerialsForIncidentRequest, stream sapb.StorageAuthority_SerialsForIncidentServer) error {
+	rows, err := ssa.dbMap.Db.QueryContext(stream.Context(), fmt.Sprintf("SELECT * FROM %s", req.IncidentTable))
```

Comment:
Since we're interpolating `req.IncidentTable`, let's sanitize it: check that it includes only A-Z, a-z, and _, and that it's less than 100 bytes. That prevents e.g. a compromised WFE from pivoting to running arbitrary DB commands.

Let's also make that change retroactively to IncidentsForSerial.

Also, since we expect a specific set of fields in a specific order below, this should enumerate those fields in the same order (`SELECT serial, regID, ...`). Typically we would just let gorp handle the mapping of names to fields; any reason not to do that here?

Reply:
> Also, since we expect a specific set of fields in a specific order below, this should enumerate those fields in the same order (`SELECT serial, regID, ...`). Typically we would just let gorp handle the mapping of names to fields; any reason not to do that here?

We're using `*sql.DB.QueryContext` instead of `*db.WrappedMap` so we can pass the context. I've gone ahead and made some additions to ensure that that's in the model matches what's in the DB.